python/ray/air/tests/test_checkpoints.py

```diff
@@ -480,6 +482,39 @@ def test_dict_cp_as_directory(self):
 
         assert not os.path.exists(checkpoint_dir)
 
+    def test_fs_cp_from_populator(self):
+        # Create checkpoint from populator
+        checkpoint = Checkpoint.from_directory_populator(
+            lambda dir: (Path(dir) / "data.txt").write_text("Data")
```

Comment:
woah I didn't you could use this syntax 😮

Reply:
wait what does this do? ``Path/"filename"`` will actually create a file object at that path??
that's kinda crazy.

Reply:
Well it will create a Path object that will create the actual file when we call `write_test()` :-)